src/useGetPosts/index.js

```diff
@@ -66,24 +66,22 @@ import { apiClient } from '../utils';
  */
 function useGetPosts( args = {}, clientId, postType ) {
 	const [ options, setOptions ] = useState( [] );
-	const [ query, setQuery ] = useState( '' );
-	const [ loading, setLoading ] = useToggle();
+	const [ query, setQuery ] = useState( [] );
+	const [ loading, setLoading ] = useToggle( true );
 	const toast = useToast();
 
 	useDeepCompareEffect( () => {
-		setLoading();
 		apiClient
 			.get( `/wp/v2/${ postType }`, { per_page: -1, post_status: 'publish', ...args } )
 			.then( ( data ) => {
 				setOptions( selectOptions( data, { id: 'value', 'title.rendered': 'label' }, [] ) );
 				setQuery( data );
-				setLoading();
 			} )
 			.catch( () => {
 				setQuery( [] );
-				setLoading();
 				toast( __( 'Error: Couldn’t retrieve posts via API.', 'sixa' ), 'error' );
-			} );
+			} )
+			.finally( setLoading );
```

Comment:
Only a native Promise (constructed with new Promise) is guaranteed to have a `.finally` method. In our case, since we are already catching any possible errors, we can use another `then` instead of using `finally`. `then` after `catch` is works like a `finally` but would always run as intended.

Reply:
Okay, thank you, I didn't know that :)